### PR TITLE
Location Service update

### PR DESCRIPTION
**Pull request by ah1053**
1. LocationService Flow similar to what discussed [here](https://mobiledgex.slack.com/archives/CKVFN0XJ5/p1619722301498800)
2. Added a complier directive to remove Unity warnings generated from CarrierInfoIntegration.cs.
3. Load MobiledgeXSettings from one place only to avoid iOS AOT errors.
4. Removing some excessive log messages from WebSocketClient and UDPClient.
5. Fixed some typos mentioned before by Bruce.
6. Social Icons
Screen Shot 2021-04-29 at 1 57 10 PM

### CODE REVIEW

**Review by franklin-huang-mobiledgex — COMMENTED** [review]
This is good. One suggestion is to save any exceptions/errors that occur when LocationService is trying to initialize and then return that exception right away in EnsureLocation(). This way the application can better handle location issues.

For example if the user rejects location permissions, then you might do this:
```
initLocationServiceException = new LocationServiceException("User rejected location permissions")
```
Then in EnsureLocation, it might look like:
```
public EnsureLocation() {
   if (initLocationServiceException != null) {
        throw initLocationServiceException
   }
   ...
}
```

File: Runtime/Scripts/LocationService.cs [hunk 1]
```diff
@@ -27,141 +27,135 @@ namespace MobiledgeX
...
         /// If Location permission is denied by User an exception will be thrown once RetrieveLocation() is called
         /// </summary>
         public static IEnumerator EnsureLocation()
         {
-            if (!SystemInfo.supportsLocationService)
+            if(!SystemInfo.supportsLocationService)
             {
                 Logger.Log("Your Device doesn't support LocationService");
-                yield return null;
+                yield break;
```
Comment (lgarner) [inline Runtime/Scripts/LocationService.cs:47]: Ah, better break.

**Review by lgarner — APPROVED** [review]
ok.

File: Runtime/Scripts/LocationService.cs [hunk 2]
```diff
@@ -27,141 +27,135 @@ namespace MobiledgeX
...
-            {
-                Input.location.Stop();
-            }
-        }
+        /// <summary>
+        /// Maximum waiting time for location services from initializing to acquiring location info in seconds (After accepting Location Permissions)
+        /// if the value selected is zero or less the default value (20 seconds) will be used.
+        /// </summary>
+        [Tooltip("Maximum waiting time for location services from initializing to acquiring location info in seconds (After accepting Location Permissions)," +
+            "if the value selected is zero or less the default value (20 seconds) will be used.")]
```
[the remark this answers is not in the document]
Comment (ah1053) [reply Runtime/Scripts/LocationService.cs:35]: I agree, Thanks

**Pull request merged**